### PR TITLE
fix(Pie): make maxRadius actually constrain percentage-based radii

### LINKED ISSUE #4885: maxRadius does not seem to have any effect in pie chart
## Do you want to request a feature or report a bug?
Bug
## What is the current behavior?
maxRadius does not change behavior of pie chart
## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar (template: https://codesandbox.io/p/sandbox/simple-line-chart-v25dl4).
https://codesandbox.io/p/sandbox/pie-chart-in-responsive-container-dmhf62

## What is the expected behavior?
chart size is limited

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
2.12.7

### PR DESCRIPTION
**Pull request by pratik-desgn**
## Description

Fixes #4885.

`<Pie maxRadius={...}>` had no visible effect. The radius that percentage-based `innerRadius`/`outerRadius` resolve against was computed in `parseCoordinateOfPie` purely from the chart's width/height (`getMaxRadius(width, height)`), completely ignoring the `maxRadius` prop.

The prop wasn't entirely dead — it's already used a few lines below to compute a *different* fallback value (for label-line positioning), which is presumably why it looked wired up on a first read. But that value is never fed back into the actual sector geometry, so setting `maxRadius` alone never changed the rendered size of the pie.

## Fix

`parseCoordinateOfPie` now prefers the explicit `pieSettings.maxRadius` over the width/height-derived default, consistent with how the prop is already used for the label-positioning fallback a few lines down.

## Test plan

- Added two tests to `test/state/selectors/pieSelectors.spec.tsx`: one asserting `outerRadius` resolves against an explicit `maxRadius` (`"80%"` of `maxRadius=100` → `80`), one confirming the previous width/height-derived fallback still applies when `maxRadius` is omitted.
- Full `test/polar` and `test/state` suites pass (953 tests, no regressions).
- `npm run lint` passes on changed files.

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated pie chart radius calculations to honor an explicitly configured maximum radius (including when set to `0`).
  * Refined percentage-based inner/outer radius resolution so values are correctly capped by the configured maximum.

* **Tests**
  * Expanded selector test coverage for pie sectors with and without a configured maximum radius, including percentage handling and the `0` max-radius edge case.

### CODE REVIEW

File: src/polar/Pie.tsx [hunk 1]
```diff
@@ -555,7 +555,7 @@ const getOuterRadius = (
 
 const parseCoordinateOfPie = (pieSettings: PieSettings, offset: ChartOffsetInternal, dataPoint: any): PieCoordinate => {
   const { top, left, width, height } = offset;
-  const maxPieRadius = getMaxRadius(width, height);
+  const maxPieRadius = pieSettings.maxRadius || getMaxRadius(width, height);
```
[the remark this answers is not in the document]
Comment (PavelVanecek) [reply src/polar/Pie.tsx]: Do this please and add a unit test for it

**Comment by pratik-desgn** [comment]
Updated to use `??` for both fallbacks so an explicit `maxRadius={0}` is preserved, and added a regression test (`should preserve a zero maxRadius when resolving percentage radii`) in `test/state/selectors/pieSelectors.spec.tsx`. Full `test/polar` + `test/state` suites pass.

**Review by PavelVanecek — APPROVED** [review]
